compiler/rustc_hir/src/lang_items.rs

```diff
@@ -45,7 +45,16 @@ impl LanguageItems {
 
     pub fn set(&mut self, item: LangItem, def_id: DefId) {
         self.items[item as usize] = Some(def_id);
-        self.reverse_items.insert(def_id, item);
+        let preexisting = self.reverse_items.insert(def_id, item);
+
+        // This needs to be a bijection.
+        if let Some(preexisting) = preexisting {
+            panic!(
```

Comment:
```suggestion
            bug!(
```

Reply:
bug is defined in a downstream crate iirc

Reply:
yeah, rustc_middle